src/map/clif.c

```diff
@@ -13755,6 +13755,106 @@ void clif_parse_ChangePetName(int fd, struct map_session_data *sd)
 	pet->change_name(sd, RFIFOP(fd,2));
 }
 
+void clif_parse_pet_evolution(int fd, struct map_session_data *sd) __attribute__((nonnull (2)));
+/// Request to Evolve the pet (CZ_RENAME_PET) [Dastgir/Hercules]
+/// 09fb <Length>.W <EvolvedPetEggID>.W {<index>.W <amount>.W}*items
+void clif_parse_pet_evolution(int fd, struct map_session_data *sd)
+{
+	//int length = RFIFOW(fd, 2);
+	int petEgg = RFIFOW(fd, 4);
+	int i = 0, idx;
+
+	if (sd->status.pet_id == 0) {
+		clif->petEvolutionResult(fd, PET_EVOL_NO_CALLPET);
+		return;
+	}
+
+	for (idx = 0; idx < MAX_INVENTORY; idx++) {
```

Comment:
this search seem to be happening in few places, better split it into it own function